infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -80,10 +80,10 @@ public CorfuServerNode(@Nonnull ServerContext serverContext) {
      * @param serverMap     Server Map with all components.
      */
     public CorfuServerNode(@Nonnull ServerContext serverContext,
-                           @Nonnull Map<Class, AbstractServer> serverMap) {
+                           @Nonnull ImmutableMap<Class, AbstractServer> serverMap) {
         this.serverContext = serverContext;
         this.serverMap = serverMap;
-        router = new NettyServerRouter(new ArrayList<>(serverMap.values()));
+        router = new NettyServerRouter(serverMap, serverContext.getServerEpoch());
```

Comment:
`serverContext.getServerEpoch()` can still return 0 if it wasn't initialized. Why not just use an invalid value like -1 ?

Reply:
Since we don't bootstrap a single node layout from a client, but rather do it on startup from the `installSingleNodeLayoutIfAbsent` method, we set the node's layout, but we never seal it. So we would need to seal both the router and epoch with the singleNodeLayout's epoch before a single node can start.

Reply:
Fixed